### ISSUE TITLE
Commit conflicts unavoidable running compaction when there are many files

### ISSUE THREAD

**Issue by westonpace** [lancedb/lance#2479]
I created a dataset with 8 million rows spread across several hundred files.  I then attempted to compact this dataset.  Because 8 compaction tasks were launched in parallel they ended up conflicting with each other and the commit failed every time I tried.

**Comment by westonpace** [comment]
Duplicate of #2397
